processing/src/main/java/io/druid/segment/IndexSpec.java

```diff
@@ -79,30 +113,39 @@ public IndexSpec()
    *
    * @param bitmapSerdeFactory type of bitmap to use (e.g. roaring or concise), null to use the default.
    *                           Defaults to the bitmap type specified by the (deprecated) "druid.processing.bitmap.type"
-   *                           setting, or, if none was set, uses the default @{link BitmapSerde.DefaultBitmapSerdeFactory}
+   *                           setting, or, if none was set, uses the default {@link BitmapSerde.DefaultBitmapSerdeFactory}
    *
-   * @param dimensionCompression compression format for dimension columns, null to use the default
-   *                             Defaults to @{link CompressedObjectStrategy.DEFAULT_COMPRESSION_STRATEGY}
+   * @param dimensionCompression compression format for dimension columns, null to use the default.
+   *                             Defaults to {@link CompressedObjectStrategy#DEFAULT_COMPRESSION_STRATEGY}
    *
    * @param metricCompression compression format for metric columns, null to use the default.
-   *                          Defaults to @{link CompressedObjectStrategy.DEFAULT_COMPRESSION_STRATEGY}
+   *                          Defaults to {@link CompressedObjectStrategy#DEFAULT_COMPRESSION_STRATEGY}
+   *
+   * @param longEncoding encoding strategy for metric and dimension columns with type long, null to use the default.
+   *                     Defaults to {@link CompressionFactory#DEFAULT_LONG_ENCODING_STRATEGY}
    */
   @JsonCreator
   public IndexSpec(
       @JsonProperty("bitmap") BitmapSerdeFactory bitmapSerdeFactory,
       @JsonProperty("dimensionCompression") String dimensionCompression,
-      @JsonProperty("metricCompression") String metricCompression
+      @JsonProperty("metricCompression") String metricCompression,
```

Comment:
Nevermind, added the toString and fromString for the enums and replaced the string with enum values. There seems to be a special handle for Uncompressed strategy for dimension compression, where getStrategy returns null for uncompressed. The caller handles the null as indicator to not use compressedIndexedInt, I think it's doing something similar to none strategy in this PR for performance reasons. I removed the null return value and did the special handling for UNCOMPRESSED strategy instead, this syntax change should not affect anything.